idmltest/src/main/scala/io/idml/test/Runner.scala

```diff
@@ -0,0 +1,193 @@
+package io.idml.test
+import java.io.File
+import java.net.URL
+import java.nio.file.{Path, Paths, StandardOpenOption}
+
+import cats._
+import cats.data.{EitherT, NonEmptyList}
+import cats.implicits._
+import cats.effect._
+import com.google.re2j.Pattern
+import io.circe.parser.{parse => parseJson}
+import io.circe.yaml.parser.{parse => parseYaml}
+import io.circe.{Decoder, Json}
+import io.circe.generic.auto._
+import io.circe.syntax._
+import io.idml.{FunctionResolverService, PluginFunctionResolverService, Ptolemy, PtolemyConf, PtolemyJson, StaticFunctionResolverService}
+import fs2._
+import gnieh.diffson.circe._
+
+import scala.util.Try
+import scala.collection.JavaConverters._
+
+class TestUtils[F[_]: Sync] {
+  def readAll(p: Path): F[String]   = fs2.io.file.readAll[F](p, 2048).through(fs2.text.utf8Decode[F]).compile.foldMonoid
+  def parseJ(s: String): F[Json]    = Sync[F].fromEither(parseJson(s))
+  def parseY(s: String): F[Json]    = Sync[F].fromEither(parseYaml(s))
+  def as[T: Decoder](j: Json): F[T] = Sync[F].fromEither(j.as[T])
+  def refToPath(parent: Path, r: Ref): F[Path] = Sync[F].fromTry(
+    Try { parent.toAbsolutePath.getParent.resolve(r.`$ref`) }
+  )
+  def writeAll(p: Path)(s: Stream[F, String]): F[Unit] =
+    s.through(fs2.text.utf8Encode[F]).to(fs2.io.file.writeAll(p, List(StandardOpenOption.TRUNCATE_EXISTING))).compile.drain
+  def print(a: Any): F[Unit]         = Sync[F].delay { println(a) }
+  def red[T <: Any](t: T): F[Unit]   = print(fansi.Color.Red(t.toString))
+  def green[T <: Any](t: T): F[Unit] = print(fansi.Color.Green(t.toString))
+  def blue[T <: Any](t: T): F[Unit]  = print(fansi.Color.Cyan(t.toString))
+}
+
+class Runner(dynamic: Boolean, plugins: Option[NonEmptyList[URL]]) extends TestUtils[IO] with CirceEitherEncoders {
+
+  def load(test: Path): IO[Tests] = readAll(test).flatMap(parseJ).flatMap(as[Tests])
+  def resolve(path: Path, tests: Tests): IO[List[ResolvedTest]] =
+    tests.tests.traverse(
+      _.resolve(
+        refToPath(path, _).flatMap(readAll),
+        refToPath(path, _).flatMap(readAll).flatMap(parseJ)
+      ))
+  def updateResolve(path: Path, tests: Tests): IO[List[UpdateableResolvedTest]] =
+    tests.tests.traverse(
+      _.updateResolve(
+        refToPath(path, _).flatMap(readAll),
+        refToPath(path, _).flatMap(readAll).flatMap(parseJ)
+      ))
+
+  def ptolemy: IO[Ptolemy] = IO {
+    val baseFunctionResolver =
+      new StaticFunctionResolverService((new DeterministicTime() :: StaticFunctionResolverService.defaults.asScala.toList).asJava)
```

Comment:
I think we need a way to feed the time in here so we're not always testing against 1970-01-01@00:00.